Enterprise v0.25 Docs + Release Notes

jwitz opened the pull request:
This PR encompasses all of the documentation changes needed for the latest LTS release of Astronomer. 

A lot of files have been modified in this PR, but only a few have new content beyond hyperlink changes. Those documents are:

- enterprise/v0.25/09_resources/01_release-notes.md (@vishwas-astro and @danielhoherd should review)
- enterprise/v0.25/06_manage-astronomer/06_upgrade-to-0-25.md ( @danielhoherd and @rbankston should review)
- enterprise/v0.25/04_deploy/04_deploy-nfs.md (@jedcunningham, @kaxil,  @andriisoldatenko, and @fhoda should review)
- enterprise/v0.25/04_deploy/02_configure-deployment.md (@paolaperaza should review)
- enterprise/v0.25/04_deploy/03_deploy-cli.md (@paolaperaza should review)

*Reviewers need to review only the docs you've been tagged with, but you're welcome to look at the others as well.*

paolaperaza commented on enterprise/v0.25/04_deploy/02_configure-deployment.md:
```diff
@@ -0,0 +1,176 @@
+
+## Configure DAG Deployment Methods
+
+Be default, Astronomer users can deploy DAGs from a local directory to an Airflow Deployment using the Astronomer CLI. This workflow is described in [Deploy via the CLI](/docs/enterprise/v0.25/deploy/deploy-cli).
+
+In an image-based Deployment, your DAGs are built directly into the Docker image that's used to run your Airflow Deployment. DAGs are rebuilt into the image each time you run `astro deploy` from the Astronomer CLI.
+
+Once you create a local Airflow project directory, this deployment method requires no additional setup. We recommend an image-based setup when getting started with Airflow.
+
+Alternatively, you can enable NFS Volume-based DAG deployment. In an NFS volume-based Deployment, you deploy DAGs to Airflow by adding them to an external NFS volume. Deploying to an NFS volume does not require rebuilding your Docker image, which makes this a good strategy for users who want to quickly or continuously deploy DAGs to Airflow.
+
+This method must be explicitly enabled on your platform. You must also create an external NFS volume and specify the location of the volume in Astronomer. For more information, read [Deploy to an NFS Volume](/docs/enterprise/v0.23/deploy/deploy-nfs).
```
Nice, good start here. A few thoughts:

- What do you think about explicitly listing the two deployment methods to start, and then explaining each one from there?
- After I jump to the second paragraph, I actually don't realize that "image-based Deployment" is what you described in the first paragraph. Maybe no need so split out description of the first method into 3 paragraphs?
- I think a screenshot of this "choice" would be nice if we have one (I tbh haven't seen it, but I presume when a user creates a deployment they pick)

jwitz replied:
I decided not to start out with the two choices, because NFS Volumes is not on by default. I think we first want to explain image-based Deployments, since that's what most people will be using, and then offer NFS as an alternative. 

Also, what are your thoughts on sayin "image-based deploys" vs. "image-based deployments"? I kind of like the first more because it differentiates from Airflow Deployments.

paolaperaza commented on enterprise/v0.25/06_manage-astronomer/04_configure-nfs.md:
```diff
@@ -0,0 +1,49 @@
+---
+title: "Configure NFS Volume-based DAG Deployment"
+navTitle: "Configure NFS Volumes"
```
Going back and forth on the title. Should it be something closer to "Deploy Code via x"? Wondering how we'll frame this/position it next to the "Deploy via CLI" doc.

paolaperaza commented on enterprise/v0.25/06_manage-astronomer/04_configure-nfs.md:
```diff
@@ -0,0 +1,49 @@
+
+## Overview
+
+Starting in Astronomer Enterprise v0.25, you can use an external NFS Volume to deploy DAGs to an Airflow environment.
```
```suggestion
Starting in Astronomer Enterprise v0.25, you can use an external NFS Volume to deploy DAGs to an Airflow Deployment on Astronomer.
```

paolaperaza replied:
Tbh I don't know what NFS means, lol - might be nice to specify 👀  Or just more generally maybe 1 sentence on what a "volume" even is and what this method is doing under the hood, for context?

paolaperaza commented on enterprise/v0.25/06_manage-astronomer/04_configure-nfs.md:
```diff
@@ -0,0 +1,49 @@
+
+Starting in Astronomer Enterprise v0.25, you can use an external NFS Volume to deploy DAGs to an Airflow environment.
+
+Unlike [deploying DAGs via the CLI](/docs/enterprise/v0.25/deploy/deploy-cli.), deploying DAGs to an NFS volume does not require rebuilding your Airflow Deployment. When a DAG is added to an NFS volume, it automatically appears in your Deployment without any further build steps.
```
```suggestion
Unlike [deploying DAGs via the Astronomer CLI](/docs/enterprise/v0.25/deploy/deploy-cli.), deploying DAGs to an NFS volume does not require rebuilding a Docker image and thus restarting the underlying Airflow service. When a DAG is added to an NFS volume, it automatically appears in the Airflow UI without requiring additional action or causing downtime.
```

paolaperaza commented on enterprise/v0.25/06_manage-astronomer/04_configure-nfs.md:
```diff
@@ -0,0 +1,49 @@
+
+This guide provides the necessary setup for connecting an NFS volume to Astronomer as a DAG deployment option. Some considerations before completing this setup:
+
+- NFS volumes are used only for DAG deploys. To add dependencies or other requirements to your Deployment, you still need to rebuild your image via the CLI as described in [Customize Images](/docs/enterprise/v0.25/develop/customize-image).
```
```suggestion
- NFS volumes are used only for DAG deploys. To add dependencies or other requirements to your Deployment, you still need to set them in your `requirements.txt` and `packages.txt` files and rebuild your image via the CLI as described in [Customize Images](/docs/enterprise/v0.25/develop/customize-image).
```

paolaperaza commented on enterprise/v0.25/06_manage-astronomer/04_configure-nfs.md:
```diff
@@ -0,0 +1,49 @@
+- NFS volumes are used only for DAG deploys. To add dependencies or other requirements to your Deployment, you still need to rebuild your image via the CLI as described in [Customize Images](/docs/enterprise/v0.25/develop/customize-image).
+- DAGs must be deployed directly to NFS volumes. They cannot be deployed via the Astronomer CLI or an Astronomer service account.
+
+## Step 1: Enable the Feature on Your Platform
```
```suggestion
## Step 1: Enable the Feature on Astronomer
```

On Astronomer? 🤷‍♀️

paolaperaza commented on enterprise/v0.25/06_manage-astronomer/04_configure-nfs.md:
```diff
@@ -0,0 +1,49 @@
+* A directory for DAGS.
+* Read access for a user with GID `50000` and UID `50000`
+
+We recommend using a GCP Filestore instance for your NFS Volume. For more information on configuring a Filestore instance, read the [GCP documentation](https://cloud.google.com/filestore/docs/creating-instances).
```
Awesome we have an example in here - think this makes it come to life. For non-GCP users, what do we recommend? For AWS is it an S3 bucket?

jwitz replied:
I think bucket-based Deployment is a separate feature that's coming in a future release. The equivalent here would be [EFS Storage](https://aws.amazon.com/premiumsupport/knowledge-center/eks-persistent-storage/), but I'm not sure whether we've tested with that yet or not. Was gonna ask about this during tech review.

paolaperaza commented on enterprise/v0.25/06_manage-astronomer/04_configure-nfs.md:
```diff
@@ -0,0 +1,49 @@
+2. Select **NFS Volume Mount**.
+3. In the **NFS Location** field that appears, enter the location of your volume-based DAG directory as `<IP>:/<path>` (for example: `192.168.0.1:/path/to/your/dags`).
+4. Save your changes to the Deployment.
+
```
Screenshot?

paolaperaza commented on enterprise/v0.25/06_manage-astronomer/07_upgrade-to-0-23.md:
```diff
@@ -0,0 +1,147 @@
+---
+title: "Upgrade to Astronomer Enterprise v0.23"
+navTitle: "Upgrade to v0.23"
```
Presume we'll modify dis to v0.25 - skipping for rn

paolaperaza reviewed (commented):
Cool cool cool @jwitz ! Some initial/directional comments here, but looks great - will let you finish it off and come back for a more thorough review.

A general thought: Did you consider adding info on NFS volume mount mechanism to our existing ["Deploy via CLI" doc](https://www.astronomer.io/docs/enterprise/v0.23/deploy/deploy-cli)? It's actually not that long, and I think it's nice to present them together for contrast. With that said, the focus is also kind of nice. Curious on your thoughts.

jwitz commented:
@paolaperaza The NFS components of this doc are ready for another look. I implemented our thoughts on the doc organization, and I also added this information to the CLI Reference guide. Let me know what you think!

paolaperaza commented on enterprise/v0.25/04_deploy/02_configure-deployment.md:
```diff
@@ -0,0 +1,176 @@
+
+## DAG Deploy Methods
+
+By default, Astronomer users can deploy DAGs by building them directly into a Docker image and deploying the image using the Astronomer CLI. This workflow is described in [Deploy via the CLI](/docs/enterprise/v0.25/deploy/deploy-cli).
+
+In this image-based deploy, your DAGs are built directly into the Docker image that's used to run your Airflow Deployment. DAGs are rebuilt into the image each time you run `astro deploy` from the Astronomer CLI.
```
```suggestion
By default, Astronomer users can deploy DAGs to an Airflow Deployment on Astronomer by building them into a Docker image and pushing that image to the Astronomer Registry via the CLI or API. This workflow is described in [Deploy via the CLI](/docs/enterprise/v0.25/deploy/deploy-cli).

When this image-based deploy mechanism is used, your DAGs are built into a Docker image alongside all other files in your Airflow project directory - your Python and OS-level packages, your Dockerfile, and your plugins. This image is then used to generate a set of Docker containers for each of Airflow's core components. Every time you run `astro deploy` via the Astronomer CLI, your DAGs are rebuilt into a new Docker image and all Docker containers are restarted.
```

paolaperaza commented on enterprise/v0.25/04_deploy/02_configure-deployment.md:
```diff
@@ -0,0 +1,176 @@
+
+In this image-based deploy, your DAGs are built directly into the Docker image that's used to run your Airflow Deployment. DAGs are rebuilt into the image each time you run `astro deploy` from the Astronomer CLI.
+
+Once you create a local Airflow project directory, this deployment method requires no additional setup. We recommend an image-based setup when getting started with Airflow.
```
Hm don't totally understand the first sentence here. Maybe rephrase to:

> Since it does not require additional setup, we recommend the image-based deploy mechanism for those getting started with Airflow.

paolaperaza commented on enterprise/v0.25/04_deploy/02_configure-deployment.md:
```diff
@@ -0,0 +1,176 @@
+
+Once you create a local Airflow project directory, this deployment method requires no additional setup. We recommend an image-based setup when getting started with Airflow.
+
+Alternatively, you can enable NFS volume-based DAG deployments. In an NFS volume-based deploy, you deploy DAGs to Airflow by adding them to an external NFS volume. Deploying to an NFS volume does not require rebuilding your Docker image, which makes this a good strategy for users who want to quickly or continuously deploy DAGs to Airflow.
+
+To use NFS volume-based deploy, you must first enable the feature on your platform and configure an external NFS volume. For more information, read [Deploy DAGs via NFS Volume](/docs/enterprise/v0.23/deploy/deploy-nfs).
```
I still just feel like this section is a bit run-on. Why not start introduce it by stating that there are 2 deploy mechanisms, and then going into describing each one?

jwitz replied:
Just made this change above.

paolaperaza commented on enterprise/v0.25/04_deploy/02_configure-deployment.md:
```diff
@@ -0,0 +1,176 @@
+
+Once you create a local Airflow project directory, this deployment method requires no additional setup. We recommend an image-based setup when getting started with Airflow.
+
+Alternatively, you can enable NFS volume-based DAG deployments. In an NFS volume-based deploy, you deploy DAGs to Airflow by adding them to an external NFS volume. Deploying to an NFS volume does not require rebuilding your Docker image, which makes this a good strategy for users who want to quickly or continuously deploy DAGs to Airflow.
```
```suggestion
For advanced teams who deploy DAG changes more frequently, Astronomer also supports an [NFS volume-based](https://kubernetes.io/docs/concepts/storage/volumes/#nfs) deploy mechanism. In this method, users deploy DAGs to an Airflow Deployment on Astronomer by adding the corresponding Python files to a shared file system such as [GCP Filestore](https://cloud.google.com/filestore/docs/creating-instances) or [AWS EFS](https://docs.aws.amazon.com/efs/latest/ug/getting-started.html). Deploying to an NFS volume does not require rebuilding a Docker image and restarting Docker containers, limiting downtime and enabling continuous deployment.
```

paolaperaza replied:
I think more accurate language here - lmk what you think.

paolaperaza commented on enterprise/v0.25/04_deploy/02_configure-deployment.md:
```diff
@@ -0,0 +1,176 @@
+
+Alternatively, you can enable NFS volume-based DAG deployments. In an NFS volume-based deploy, you deploy DAGs to Airflow by adding them to an external NFS volume. Deploying to an NFS volume does not require rebuilding your Docker image, which makes this a good strategy for users who want to quickly or continuously deploy DAGs to Airflow.
+
+To use NFS volume-based deploy, you must first enable the feature on your platform and configure an external NFS volume. For more information, read [Deploy DAGs via NFS Volume](/docs/enterprise/v0.23/deploy/deploy-nfs).
```
```suggestion
To use NFS volume-based deploy, you must first enable the feature at the platform-level and configure an external NFS volume for every Airflow Deployment you'd like to apply this mechanism to. For more information, read [Deploy DAGs via NFS Volume](/docs/enterprise/v0.23/deploy/deploy-nfs).
```

jwitz replied:
I decided to cut the second part about actually using the mechanism in production. As long as they know it's not on by default, they can go to the next guide to see the actual details of implementing this.

jwitz commented on enterprise/v0.25/06_manage-astronomer/06_upgrade-to-0-25.md:
```diff
@@ -0,0 +1,146 @@
+---
```
All I did to this doc was remove the breaking changes considerations and change all of the links and scripts to be 0.25-specific. Let me know if this upgrade differs in any way from the v0.23 upgrade.

vishwas-astro replied:
@jwitz - @danielhoherd is still working on the upgrade scripts. We should probably create the upgrade part as a separate PR.

jwitz commented on enterprise/v0.25/09_resources/01_release-notes.md:
```diff
@@ -0,0 +1,31 @@
+
+### Bug Fixes
+
+- Fixed an issue where Grafana pods would not run properly when `global.ssl.mode` was set to `prefer`. ([Source](https://github.com/astronomer/astronomer/pull/1082))
```
This was the only non-NFS related, customer-facing change I could find for v0.25. Do we have any more? @vishwas-astro

vishwas-astro replied:
Actually this got pulled out of 0.25.0 and moved to 0.25.1. So you can remove it.

vishwas-astro commented:
@jwitz - I think @jedcunningham and @kaxil should also review the "nfs volume based dag deploys" related sections of this PR.

vishwas-astro commented on enterprise/v0.25/02_develop/01_cli-quickstart.md:
````diff
@@ -0,0 +1,264 @@
+To install a specific version of the Astronomer CLI, specify `-s -- major.minor.patch` as a flag at the end of the cURL command. To install v0.23.2, for example, run:
+
+```
+$ curl -sSL https://install.astronomer.io | sudo bash -s -- v0.23.2
````
Should we change this to 0.25.0 instead of 0.23.2?

jwitz replied:
I think we tried writing this section so that we wouldn't need to update it with each release. The `for example` implies that you could replace this with any relevant version.

vishwas-astro commented on enterprise/v0.25/02_develop/01_cli-quickstart.md:
````diff
@@ -0,0 +1,264 @@
+```
+Astro CLI Version: 0.23.2
+Git Commit: c4fdeda96501ac9b1f3526c97a1c5c9b3f890d71
+```
````
Should we change this to 0.25.0 and its Git Commit SHA?

jwitz replied:
See above comment - we shouldn't need to replace these with each release.

vishwas-astro commented on enterprise/v0.25/02_develop/01_cli-quickstart.md:
```diff
@@ -0,0 +1,264 @@
+$ astro version
+Astro CLI Version: 0.23.2
+Astro Server Version: 0.23.9
+Git Commit: 748ca2e9de1e51e9f48f9d85eb8315b023debc2f
```
Check the above output inn 0.25.0 once before putting this up.

jwitz replied:
See above comment - we shouldn't need to replace these with each release.

vishwas-astro commented on enterprise/v0.25/04_deploy/04_deploy-nfs.md:
```diff
@@ -0,0 +1,58 @@
+Any NFS volume can be used for this step as long as it has:
+
+* A directory for DAGs.
+* Read access for a user with GID `50000` and UID `50000`.
```
@jwitz do we expect our users to know how to do this?

jwitz replied:
Not on their own, but it's a process that takes place off of the Astronomer platform and could vary based on what NFS Volume solution they're using. We link out to the documentation for each supported cloud's NFS volume solution, so I expect users to go there to determine what specific steps need to be completed for this prerequisite.

vishwas-astro replied:
Not a blocker for this PR, but maybe link out to an example for how it is done with one of the NFS volume solutions.

vishwas-astro commented on enterprise/v0.25/04_deploy/04_deploy-nfs.md:
```diff
@@ -0,0 +1,58 @@
+
+    ![Custom Release Name Field](https://assets2.astronomer.io/main/docs/astronomer-ui/nfs.png)
+
+4. In the **NFS Location** field that appears, enter the location of your volume-based DAG directory as `<IP>:/<path>` (for example: `192.168.0.1:/path/to/your/dags`).
```
I see the note below in this doc, but we should probably update CLI docs as well.

jwitz replied:
CLI reference guide is updated in this PR. I just added some information on the flags that are nearly identical to what's in the help text. As for our CLI quickstart, I added a link to the document in the "Next Steps" section. I think that's all we should need since the feature is off by default. 

Because we have the full formatting example here, and because the UI indicates that nfs:// is prepended, I don't think it's necessary to include that note in our docs. We needed it in the CLI help text because there was nowhere else to indicate that this would be the case.

vishwas-astro commented on enterprise/v0.25/06_manage-astronomer/05_upgrade-astronomer-patch.md:
```diff
@@ -0,0 +1,86 @@
+- Y: Minor Version
+- Z: Patch/Hotfix
+
+For example, upgrading Astronomer from v0.16.4 to v0.16.5 would be considered upgrading to a patch version, whereas upgrading from v0.15.0 to v0.16.0 would be considered upgrading to the latest minor version.
```
Do you want to change this example to 0.25.* specific patch versions?

jwitz replied:
Similar to before, we could but I think it makes sense to leave these as examples that don't change version. It would increase our documentation maintenance if we update every versioning example throughout our docs, and I think users will be able to extrapolate and apply this example to 0.25.

vishwas-astro commented on enterprise/v0.25/06_manage-astronomer/06_upgrade-to-0-25.md:
````diff
@@ -0,0 +1,146 @@
+kubectl delete -f https://raw.githubusercontent.com/astronomer/astronomer/master/bin/migration-scripts/lts-to-lts/0.23-to-0.25/manifests/upgrade-0.23-to-0.25.yaml
+```
+
+## Step 9: Upgrade the Astronomer CLI to v0.25
````
Not sure if these need to be specific patch version of CLI. Ex: 0.25.0

@andriisoldatenko, wdyt?

vishwas-astro commented on enterprise/v0.25/06_manage-astronomer/06_upgrade-to-0-25.md:
````diff
@@ -0,0 +1,146 @@
+To do so via Homebrew, run:
+
+```sh
+$ brew install astronomer/tap/astro@0.25
````
Also here, does it need to be 0.25.0? @andriisoldatenko

paolaperaza replied:
@vishwas-astro This pulls the latest `0.25.0` patch, which I think is what we want

vishwas-astro commented:
@jwitz do we not have any cloud release notes for 0.25.0 at all? In all these docs that are getting updated, i did not notice release notes for cloud.

jwitz commented:
@vishwas-astro Cloud release notes are available for review here: https://github.com/astronomer/docs/pull/344

I wrote something similar to what we had for v0.23.9, which was another Enterprise-only release.

paolaperaza commented on enterprise/v0.25/04_deploy/04_deploy-nfs.md:
```diff
@@ -0,0 +1,58 @@
+
+## Provision an NFS Volume
+
+While you can use any NFS volume for this step, we recommend using each support cloud provider's own NFS volume solution:
```
typo here?

jwitz replied:
Yeah it was originally supposed to be "each supported", but I like this wording better.

paolaperaza replied:
Cool, confirmed this change

paolaperaza commented on enterprise/v0.25/06_manage-astronomer/06_upgrade-to-0-25.md:
````diff
@@ -0,0 +1,146 @@
+$ brew install astronomer/tap/astro@0.25
+```
+
+All team members within your organization should upgrade the Astronomer CLI individually before taking any further action on the platform or in a local Airflow environment. For a detailed breakdown of CLI changes between versions, refer to [Astronomer CLI releases](https://github.com/astronomer/astro-cli/releases). For detailed install guidelines and more information on the Astronomer CLI, refer  ...
````
Stating this more strongly - given the NFS volume feature, pretty sure deploys with earlier versions of the Astro CLI won't work with v0.25

jwitz commented on enterprise/v0.25/02_develop/01_cli-quickstart.md:
```diff
@@ -0,0 +1,266 @@
+After installing and trying out the Astronomer CLI, we recommend reading through the following guides:
+
+* [Astronomer CLI Reference Guide](/docs/enterprise/v0.25/resources/cli-reference)
+* [Deploy DAGs Using the Astronomer CLI](/docs/enterprise/v0.25/deploy/deploy-cli/)
```
The title for the doc itself has "Using" so I updated that to "via" as well.

jwitz commented on enterprise/v0.25/04_deploy/03_deploy-cli.md:
```diff
@@ -0,0 +1,95 @@
+
+If you've used the Astronomer CLI to develop locally, you'll find it similarly easy to deploy your DAGs to an Airflow Deployment on Astronomer.
+
+This guide provides the setup steps for deploying DAGs to Astronomer using the Astronomer CLI. In this workflow, DAGs are built directly into a Docker image and pushed to a given Airflow Deployment on Astronomer.
```
Moved some of the sentences around a bit, but otherwise implemented this.

paolaperaza replied:
Had a follow up comment on this section in my most recent review, but saw the changes here & resolving!

jwitz commented on enterprise/v0.25/04_deploy/03_deploy-cli.md:
```diff
@@ -0,0 +1,95 @@
+
+This guide provides the setup steps for deploying DAGs to Astronomer using the Astronomer CLI. In this workflow, DAGs are built directly into a Docker image and pushed to a given Airflow Deployment on Astronomer.
+
+For guidance on automating this process, refer to [Deploy to Astronomer via CI/CD](/docs/enterprise/v0.25/deploy/ci-cd/). To learn how to deploy other components to Astronomer, such as dependencies or custom images, read [Customize your Image](/docs/enterprise/v0.25/develop/customize-image/).
```
Yeah, I think components should refer only to Scheduler, Webserver, etc for simplicity. Simplified this language a bit.

paolaperaza replied:
Change applied in different commit - thanks @jwitz !

jwitz commented on enterprise/v0.25/06_manage-astronomer/06_upgrade-to-0-25.md:
````diff
@@ -0,0 +1,146 @@
+- **Kubernetes**: Your version must be greater than or equal to 1.14 and less than 1.19. If you need to upgrade Kubernetes, contact your cloud provider's support or your Kubernetes administrator.
+- **Airflow Images**: You must be using an Astronomer Certified Airflow image, and the version of your image must be 1.10.5 or greater. In addition, your image should be in the following format:
+```
+astronomerinc/ap-airflow:<airflow-version>-<build-number>-<distribution>-onbuild
````
Does this need to be updated in the v0.23 guide as well? 😬

paolaperaza replied:
Less of a big deal, but yes - I'd say so 😬

jwitz commented on enterprise/v0.25/06_manage-astronomer/06_upgrade-to-0-25.md:
````diff
@@ -0,0 +1,146 @@
+```
+For example, all of the following images would work for this upgrade:
+```sh
+astronomerinc/ap-airflow:1.10.12-1-alpine3.10-onbuild
+astronomerinc/ap-airflow:1.10.12-1-alpine3.10
+astronomerinc/ap-airflow:1.10.5-9-buster-onbuild
+astronomerinc/ap-airflow:1.10.5-9-buster
````
See above, should these be changed in our Upgrade to v0.23 doc as well?

paolaperaza replied:
Responded above! Resolving this comment for now

jwitz commented on enterprise/v0.25/06_manage-astronomer/06_upgrade-to-0-25.md:
```diff
@@ -0,0 +1,146 @@
+
+## Step 2: Check Permissions
+
+Minor version upgrades can be initiated only by a user with System Admin permissions on Astronomer. To confirm you're an Astronomer SysAdmin, confirm that you have access to **System Admin** features in the in the Astronomer UI:
```
This was in the v0.23 version as well. Updated it there too

jwitz commented:
@paolaperaza This should be good to ship! One outstanding question about whether a change needs to be backported to v0.23, but otherwise everything's been updated.

paolaperaza commented on enterprise/v0.25/04_deploy/03_deploy-cli.md:
```diff
@@ -0,0 +1,95 @@
+
+This guide provides the setup steps for deploying DAGs to Astronomer using the Astronomer CLI.
+
+If you've used the Astronomer CLI to develop locally, you'll find it similarly easy to deploy your DAGs to an Airflow Deployment on Astronomer. The Astronomer CLI builds your DAGs into a Docker image alongside all other files in your Airflow project directory, including your Python and OS-level packages, your Dockerfile, and your plugins. The resulting image is then used to generate a set of Dock ...
```
```suggestion
If you've used the Astronomer CLI to develop locally, you'll find it similarly easy to deploy your DAGs to an Airflow Deployment on Astronomer. The Astronomer CLI builds your DAGs into a Docker image alongside all other files in your Airflow project directory, including your Python and OS-level packages, your Dockerfile, and your plugins. The resulting image is then used to generate a set of Docker containers for each of Airflow's core components.
```

These components are basically containers/pods for the Scheduler, Webserver, etc. - wouldn't say they're "pushed to Astronomer", they're just created within the Astro namespace from the docker image (that's the thing that's "pushed"). I don't think we need to explain that further though - can just delete I think.

paolaperaza reviewed (approved):
Alrighty - did a final pass and these look good to me @jwitz ! Responded to your comment about v0.23 (yes, I think should be backported) and then left one other minor copy suggestion that I'll defer to you on.
